src/compiler/interface2.jl

```diff
@@ -32,3 +43,59 @@ end
   back = slots!(inlineable!(back))
   return update!(meta.code, back)
 end
+
+
+
+
+#=="""
+  chainrules_blacklist(f, args...,)
+
+This is used to disable the use of ChainRule's definitions
+for particular functions/methods.
+
+It is not required if a Zygote rule has already been defined directly.
+"""==#
+chainrules_blacklist(f, args...) = false
+
+# ChainRules does higher-order functions badly
+# see https://github.com/JuliaDiff/ChainRules.jl/issues/122
+for f in (map, broadcast, mapreduce, mapfoldl, mapfoldr)
+    @eval chainrules_blacklist(::typeof($f), args...) = true
+end
+chainrules_blacklist(::typeof(sum), f, x::AbstractArray{<:Real}) = true
+# Except for sum(abs2, xs), that is fine
+chainrules_blacklist(::typeof(sum), ::typeof(abs2), x::AbstractArray{<:Real}) = false
+
+# ChainRules current Wirtinger deriviative is not compatible
+# reconsider after https://github.com/JuliaDiff/ChainRulesCore.jl/pull/29
+for f in (abs, abs2, conj, adjoint, hypot, angle, imag, real)
+  @eval chainrules_blacklist(::typeof($f), ::Complex) = true
+end
+
+# Sum of nonarrays doesn't really work
+# Fixed in https://github.com/JuliaDiff/ChainRules.jl/pull/124
+chainrules_blacklist(::typeof(sum), x) = true
+chainrules_blacklist(::typeof(sum), x::AbstractArray{<:Real}) = false
+
+
+#=="""
+  _pullback_via_chainrules(pb)
+
+Converts a ChainRules pullback into a Zygote pullback.
+`pb` should be a ChainRules pullback, as returned from the second return value of `rrule`
+"""==#
+function _pullback_via_chainrules(pb)
+  # This is the optimized version of
+  # _pullback_via_chainrules(pb) = (Δs...) -> zextern.(pb(Δs...))
+  function zback(Δs...)
+    ∂s = pb(Δs...)
+    ntuple(length(∂s)) do ii
+      ∂ = ∂s[ii]
+      zextern(∂)
+    end
+  end
+end
+
+zextern(x) = ChainRules.extern(x)
```

Comment:
Shouldn't this be:

```suggestion
zextern(x) = ChainRules.extern(conj(x))
```

should also add test for that